cmd/livepeer/livepeer.go

```diff
@@ -640,11 +641,17 @@ func main() {
 			if err != nil {
 				glog.Fatal("Error setting orch webhook URL ", err)
 			}
-			n.OrchestratorPool = discovery.NewWebhookPool(n, whurl)
+			n.OrchestratorPool = discovery.NewWebhookPool(whurl)
 		} else if len(orchURLs) > 0 {
-			n.OrchestratorPool = discovery.NewOrchestratorPool(n, orchURLs)
+			n.OrchestratorPool = discovery.NewOrchestratorPool(orchURLs)
 		} else if *network != "offchain" {
-			n.OrchestratorPool = discovery.NewDBOrchestratorPoolCache(n)
+			dbOrchPoolCache := discovery.NewDBOrchestratorPoolCache(dbh, n.Eth, roundsWatcher, n.Sender)
+			bcast := core.NewBroadcaster(n)
+			dbOrchPoolCache.PollOrchestratorInfo(bcast)
```

Comment:
If `PollOrchestratorInfo()` caches off-chain info upon being called (see my comment in this review specifically on `PollOrchestratorInfo()` further down), then I think we should call the method after we call `CacheTranscoderPool()` to make sure that `PollOrchestratorInfo()` has access to an up-to-date list of active set members.

Reply:
I don't think so because that will lead to a double operation of fetching off-chain info for the orchestrators in the pool in the case where `originalLastSeenBlock.Cmp(currentRoundStartBlock) < 0` because  `CacheTranscoderPool`will also fetch off-chain info.

If we call it before that block we will only fetch orchestrator info for orchestrators that are already in the database through event backfilling.

Reply:
At the moment, we will fetch off-chain info for all Os included in the transcoder pool and then when we call `PollOrchestratorInfo()` we will immediately fetch off-chain info for the same Os *again*. Seems simpler to keep the act of fetching off-chain info separate from the act of caching the transcoder pool with the latter just updating addresses in the DB instead of also fetching off-chain info for all the addresses. The former would guarantee that we have an up-to-date view of the active set and *then* we can start polling for off-chain info.

Reply:
right, I see what you're saying. makes sense.

Reply:
ec531f3b directly stores O's fetched on-chain in the database when calling `CacheTranscoderPool`, rather than fetching their off-chain info as well.